### PR TITLE
Rework docstring of XLMRoberta

### PR DESCRIPTION
**Pull request by abuelnasr0**
my first  PR in keras_nlp and in open source contribution
for #867 issue I reworked docstring of XLMRoberta in a way that is similar #843
I have few questions and suggestions:
1- I changed max_Sequence_length description in xlm_roberta_backbone.py. should I write the old description back ?
2- in xlm_roberta_tokenizer.py. I suggest deleting lines 38 to 42. what do you think ?
3- I suggest adding a training method for XLMRoberta tokenizer for custom vocabulary 
4- for training sentencepair tokenizer we need to import io and scentencepice should I import them in the example code 
5- should I delete the example in xlm_roberta_preprocessor.py in lines 43 to 45
6- I suggest adding a softmax activation in XLMRoberta classifier
7- I added Arabic examples because XLMRoberta is multilingual model and it will be good to show that. what do you think ?

I made my tests in here https://colab.research.google.com/drive/1uyIQetDu9auoMP4ZHVX82zLW3tkEzZx7#scrollTo=X72rpihBvZX7

waiting for feedback.
@mattdangerw 
thanks.

### CODE REVIEW

**Comment by mattdangerw** [comment]
Thanks! This look great and really appreciate the attention to detail here.

> 1- I changed max_Sequence_length description in xlm_roberta_backbone.py. should I write the old description back ?

New description looks good. We may want to copy it for other models as a follow up.

> 2- in xlm_roberta_tokenizer.py. I suggest deleting lines 38 to 42. what do you think ?

The note about the weird sentencepiece modifications? I think we can leave it in, but maybe it would be better if the paragraph began. `Note: If you are providing you own custom SentencePiece model, ...`

> 3- I suggest adding a training method for XLMRoberta tokenizer for custom vocabulary

SGTM! Thank you

> 4- for training sentencepair tokenizer we need to import io and scentencepice should I import them in the example code

We don't generally do imports in the example code, we just assume a fixed set of symbols are available (including sentencepiece and io for now).

> 5- should I delete the example in xlm_roberta_preprocessor.py in lines 43 to 45

We can leave it in I think.

> 6- I suggest adding a softmax activation in XLMRoberta classifier

Interesting point! We have actually discussed this quite a bit. For now, we are sticking with logit output across the library (it is more the norm, and allows for manipulation of the logits when doing generation). But we may revisit this! And appreciate the feedback.

> 7- I added Arabic examples because XLMRoberta is multilingual model and it will be good to show that. what do you think ?

I like it!

File: keras_nlp/models/xlm_roberta/xlm_roberta_preprocessor.py [hunk 1]
```diff
@@ -77,77 +65,84 @@ class XLMRobertaPreprocessor(Preprocessor):
...
-    # Tokenize and a batch of single sentences.
-    sentences = tf.constant(
-        ["The quick brown fox jumped.", "Call me Ishmael."]
+    # Tokenize a batch of single sentences.
+    preprocessor(["The quick brown fox jumped.", "اسمي اسماعيل"])
+
+    # Preprocess a batch of sentence pairs.
+    # When handling multiple sequences, always convert to tensors first!
+    first = tf.constant(["The quick brown fox jumped.", "اسمي اسماعيل"])
+    second = tf.constant(["The fox tripped.", "Oh look, a whale."])
```
Comment (mattdangerw) [inline keras_nlp/models/xlm_roberta/xlm_roberta_preprocessor.py:91]: In this examples, can we replace `Oh look, a whale.` with a short Arabic sentence that could follow the first?

We want to make it clear how the batches will be handled. E.g. position 0 in both constants is for one batch sample and position 1 is another.

File: keras_nlp/models/xlm_roberta/xlm_roberta_tokenizer.py [hunk 2]
```diff
@@ -27,19 +27,19 @@
...
+    """An XLM-RoBERTa tokenizer using SentencePiece subword segmentation.
 
     This tokenizer class will tokenize raw strings into integer sequences and
     is based on `keras_nlp.tokenizers.SentencePieceTokenizer`. Unlike the
     underlying tokenizer, it will check for all special tokens needed by
     XLM-RoBERTa models and provides a `from_preset()` method to automatically
-    download a matching vocabulary for a XLM-RoBERTa preset.
+    download a matching vocabulary for an XLM-RoBERTa preset.
 
     The original fairseq implementation of XLM-RoBERTa modifies the indices of
```
Comment (mattdangerw) [inline keras_nlp/models/xlm_roberta/xlm_roberta_tokenizer.py:38]: Re your comment, we could rewrite this slightly to be shorter...

Note: If you are providing you own custom SentencePiece model, the original fairseq implementation of XLM-RoBERTa re-maps some token indices from the underlying sentencepiece output. To preserve compatibility, we do the same re-mapping here.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
This looks great! Just had a couple comments.

File: keras_nlp/models/xlm_roberta/xlm_roberta_tokenizer.py (hunk 2 above)
Comment (abuelnasr0) [reply keras_nlp/models/xlm_roberta/xlm_roberta_tokenizer.py]: @mattdangerw requested changes has been pushed.

**Review by mattdangerw — APPROVED** [review]
This LGTM! Really appreciate all the feedback you gave, this is an impressive first contribution!

**Comment by abuelnasr0** [comment]
> This LGTM! Really appreciate all the feedback you gave, this is an impressive first contribution!
Thank you!
